src/main/java/org/opentripplanner/routing/algorithm/RoutingWorker.java

```diff
@@ -63,29 +65,53 @@ public RoutingWorker(RaptorConfig<TripSchedule> config, RoutingRequest request)
     }
 
     public RoutingResponse route(Router router) {
-        try {
-            List<Itinerary> itineraries;
+        List<Itinerary> itineraries = new ArrayList<>();
+        List<RoutingError> routingErrors = new ArrayList<>();
 
+        try {
             // Direct street routing
-            itineraries = new ArrayList<>(DirectStreetRouter.route(router, request));
+            itineraries.addAll(DirectStreetRouter.route(router, request));
+        } catch (Exception e) {
```

Comment:
You should only catch business specific exceptions that can be mapped to a RoutingError.